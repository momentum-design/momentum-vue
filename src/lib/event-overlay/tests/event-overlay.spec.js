import { mount } from '@vue/test-utils';
import EventOverlay from '../index.vue';
import Button from '../../button/index.vue';
import Popover from '../../popover/index.vue';
import Vue from 'vue';

Vue.component(EventOverlay.name, EventOverlay);
Vue.component(Button.name, Button);
Vue.component(Popover.name, Popover);

global.MutationObserver = class {
  constructor(callback) {}
  disconnect() {}
  observe(element, initObject) {}
};

describe('EventOverlay', () => {
  beforeAll(() => {
    jest.clearAllTimers();
    jest.useFakeTimers();
  });

  it('should match snapshot', () => {
    const wrapper = mount(EventOverlay, {
      slots: {
        default: '<span>Test</span>',
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

});

