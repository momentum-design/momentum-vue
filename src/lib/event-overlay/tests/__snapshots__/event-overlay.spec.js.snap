// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventOverlay should match snapshot 1`] = `undefined`;
